nistats/first_level_model.py

```diff
@@ -557,3 +568,214 @@ def compute_contrast(self, contrast_def, stat_type=None,
             '%s of contrast %s' % (output_type, contrast_name))
 
         return output
+
+
+def first_level_models_from_bids(dataset_path, task_id, model_init=None,
+                                 preproc_variant=None, preproc_space=None):
+    """Create FirstLevelModel objects and fit arguments from a BIDS dataset.
+
+    Parameters
+    ----------
+    dataset_path: str
+        Directory of the highest level folder of the BIDS dataset. Should
+        contain subject folders and a derivatives folder.
+
+    task_id: str
+        Task_id as specified in the file names. Of the form _task-*_.
+
+    model_init: dict, optional
```

Comment:
The dictionary is not part of BIDS. I would say this is more an utility to avoid modifying one by one the returned models with different parameters from the default like I was doing initially. I chose a dictionary for two reasons. 1) The parameters are very numerous and irrelevant to the notion of getting the models from BIDS. 2) To avoid the problem of maintaining this function arguments if the FirstLevelModel arguments change in any way.